### PR TITLE
feat: add TagGroup component

### PR DESCRIPTION
**Pull request by thaytharma**
Add `Tag.Group` with a required `label`-prop to improve accessibility in `Tag`.

### CODE REVIEW

File: packages/dnb-design-system-portal/src/docs/uilib/components/tag/Examples.js [hunk 1]
```diff
@@ -4,28 +4,42 @@
...
 
 export const TagWithSecondaryIcon = () => (
-  <ComponentBox data-visual-test="tag-secondary-icon" scope={{ Car }}>
-    {() => /* jsx */ `
-<Tag icon={Car} text="Vehicle"/>
+  <ComponentBox data-visual-test="tag-secondary-icon" scope={{ Funds, Stocks, Equities}}>
+      {() => /* jsx */ `
+<Tag.Group label="Investments:">
+  <Tag icon={Funds} text="Funds"/>{' '}
+  <Tag icon={Stocks} text="Stocks"/>{' '}
```
Comment (langz) [inline packages/dnb-design-system-portal/src/docs/uilib/components/tag/Examples.js:40]: Are these, `{' '}`, included on purpose?

File: packages/dnb-design-system-portal/src/docs/uilib/components/tag/demos.md [hunk 2]
```diff
@@ -6,13 +6,18 @@ import {
...
-TagClickable
+TagClickable,
+TagGroupWithData,
 } from 'Docs/uilib/components/tag/Examples'
 
 ## Demos
 
 #### Tag
 
+To ensure the correct use of the Tags, we recommend using a `Tag.Group` with `Tag`-components as children. <br/>
```
Comment (langz) [inline packages/dnb-design-system-portal/src/docs/uilib/components/tag/demos.md:17]:
```suggestion
To ensure the correct use of the Tags, we require using a `Tag.Group` with `Tag`-components as children. <br/>
```

Not sure if it's required or recommended.
Based on the following lines, it seems required: https://github.com/dnbexperience/eufemia/pull/1208/files#diff-5c0faf91b07b3332d489663f41b6fdf5543c0be1e9936092bc9282e3e473a2beR99

File: packages/dnb-eufemia/src/components/tag/TagGroup.tsx [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+import classnames from 'classnames'
+
+// Components
+import { createSpacingClasses } from '../space/SpacingHelper'
+
+// Shared
+import Context from '../../shared/Context'
+import { ISpacingProps } from '../../shared/interfaces'
+import { extendPropsWithContext } from '../../shared/component-helper'
+import { Tag } from '..'
```
Comment (langz) [inline packages/dnb-eufemia/src/components/tag/TagGroup.tsx:11]:
```suggestion
import { Tag } from './Tag'
```

File: packages/dnb-eufemia/src/components/tag/TagGroup.tsx [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+import { TagProps } from './Tag'
```
Comment (langz) [inline packages/dnb-eufemia/src/components/tag/TagGroup.tsx:12]:
```suggestion
import Tag, { TagProps } from './Tag'
```

**Comment by langz** [comment]
Seems like something's happened to the screenshot tests 😂 
image

**Comment by langz** [comment]
Looks good, I believe I'll end up doing something similar to the Avatar as well.

File: packages/dnb-design-system-portal/src/docs/uilib/components/tag/demos.md (hunk 2 above)
Comment (thaytharma) [reply packages/dnb-design-system-portal/src/docs/uilib/components/tag/demos.md:17]: > Seems like something's happened to the screenshot tests 😂 image

😂 This has been updated, something is wrong with my portal..

File: packages/dnb-eufemia/src/components/tag/TagGroup.tsx (hunk 4 above)
Comment (thaytharma) [reply packages/dnb-eufemia/src/components/tag/TagGroup.tsx]: Updated!

File: packages/dnb-design-system-portal/src/docs/uilib/components/tag/Examples.js (hunk 1 above)
Comment (thaytharma) [reply packages/dnb-design-system-portal/src/docs/uilib/components/tag/Examples.js]: This has been removed and updated with the `right`(`Space`-prop) instead. 😊

File: packages/dnb-design-system-portal/src/docs/uilib/components/tag/Examples.js [hunk 5]
```diff
@@ -4,28 +4,45 @@
...
 
 import ComponentBox from 'dnb-design-system-portal/src/shared/tags/ComponentBox'
-import { car_1 as Car, send as Send } from '@dnb/eufemia/src/icons'
+import { funds as Funds } from '@dnb/eufemia/src/icons'
+import { stocks as Stocks } from '@dnb/eufemia/src/icons'
+import { equities_and_mutual_funds as Equities } from '@dnb/eufemia/src/icons'
+import { cat as Cat } from '@dnb/eufemia/src/icons'
+import { horse as Horse } from '@dnb/eufemia/src/icons'
+import { dog as Dog } from '@dnb/eufemia/src/icons'
+import { send as Send } from '@dnb/eufemia/src/icons'
```
Comment (dinarosv) [inline packages/dnb-design-system-portal/src/docs/uilib/components/tag/Examples.js:13]: Isn't there an eslint error for multiple imports here? 💭

Comment (thaytharma) [reply packages/dnb-design-system-portal/src/docs/uilib/components/tag/Examples.js]: Good catch! No errors here.. 🙁
Updated the imports 😊

File: packages/dnb-eufemia/src/components/tag/TagGroup.tsx [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+  return (
+    <TagGroupContext.Provider value={props}>
+      <div
+        className={classnames('dnb-tag--group', spacingClasses, className)}
+        data-testid="tag-group"
+        {...props}
+      >
+        <span data-testid="tag-group-label" className="dnb-sr-only">
+          {label}
+        </span>
```
Comment (dinarosv) [inline packages/dnb-eufemia/src/components/tag/TagGroup.tsx:82]: Are there cases where we would want to show the group label visually? Like a `FormLabel`.. just a thought 💭

**Review by dinarosv — APPROVED** [review]
Great work 😄  ⭐ 

Had some thoughts, but it's not necessary additions :)

File: packages/dnb-eufemia/src/components/tag/TagGroup.tsx (hunk 6 above)
Comment (thaytharma) [reply packages/dnb-eufemia/src/components/tag/TagGroup.tsx:82]: I think that is a good point!
I can discuss this with the designers. ☺️

File: packages/dnb-eufemia/src/components/tag/TagGroup.tsx [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+
+export interface TagGroupProps {
+  /**
+   * Aria label to describe the tag group
+   * Default: null
+   */
+  label: string
+
+  /**
+   * Pass in a list of your pages as objects of tags.
```
Comment (langz) [inline packages/dnb-eufemia/src/components/tag/TagGroup.tsx:21]:
```suggestion
   * Pass in a list of your tags as objects of tags.
```
Not exactly sure, but pages sounded strange 😂

**Review by langz — APPROVED** [review]

File: packages/dnb-eufemia/src/components/tag/TagGroup.tsx (hunk 7 above)
Comment (thaytharma) [reply packages/dnb-eufemia/src/components/tag/TagGroup.tsx]: Good catch, you are right😅
Updated this. Changed to `* Pass in a list of Tag objects.`

**Pull request merged**